### PR TITLE
[BUILD] Read default proto version from `third_party_release`

### LINKED ISSUE #2676: [BUILD] protoc error during build of opentelemetry-cpp with OTLP HTTP enabled
**Describe your environment** 

linux, gcc 7.3.1, protobuf 26.1, latest opentelemetry-cpp

**Steps to reproduce**

mkdir otel-cpp-starter
cd otel-cpp-starter
git clone https://github.com/open-telemetry/opentelemetry-cpp.git
cd opentelemetry-cpp
mkdir build
cd build
cmake -DBUILD_TESTING=OFF -DWITH_OTLP_HTTP=ON -DBUILD_SHARED_LIBS=OFF -DWITH_ABSEIL=ON -DCURL_LIBRARY=/BuySide_dev_home/users/procques/libcurl/lib -DCURL_INCLUDE_DIR=/home/libcurl/include -DCMAKE_PREFIX_PATH=/home/abseil/Install -DProtobuf_LIBRARIES=/tmp/protobuf/lib64/libprotobuf.a -DProtobuf_INCLUDE_DIR=/tmp/protobuf/include/ -DProtobuf_PROTOC_EXECUTABLE=/tmp/protobuf/bin/protoc ..

(works; no errors)

cmake --build .

[  5%] Built target opentelemetry-proto

[  5%] [Run]: "/tmp/protobuf/bin/protoc" "--proto_path=/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto" "--cpp_out=/home/otel-cpp-starter/opentelemetry-cpp/build/generated/third_party/opentelemetry-proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/common/v1/common.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/resource/v1/resource.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/trace/v1/trace.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/logs/v1/logs.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/metrics/v1/metrics.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/profiles/v1experimental/profiles.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/profiles/v1experimental/pprofextended.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/collector/trace/v1/trace_service.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/collector/logs/v1/logs_service.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/collector/metrics/v1/metrics_service.proto" "/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/collector/profiles/v1experimental/profiles_service.proto"

Could not make proto path relative: /home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/profiles/v1experimental/profiles.proto: No such file or directory

gmake[2]: *** [CMakeFiles/opentelemetry_proto.dir/build.make:74: generated/third_party/opentelemetry-proto/opentelemetry/proto/common/v1/common.pb.h] Error 1

gmake[1]: *** [CMakeFiles/Makefile2:694: CMakeFiles/opentelemetry_proto.dir/all] Error 2

gmake: *** [Makefile:136: all] Error 2

(The directory
/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/
exists, but
/home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/profiles
doesn't exist

**What is the expected behavior?**
I expected the build to complete and generate opentelemetry libraries, as it does when OTLP HTTP is not enabled.

**What is the actual behavior?**
The build failed because a command it executed referred to a directory which didn't exist:
Could not make proto path relative: /home/otel-cpp-starter/opentelemetry-cpp/build/opentelemetry-proto-prefix/src/opentelemetry-proto/opentelemetry/proto/profiles/v1experimental/profiles.proto: No such file or directory

**Additional context**

### PR DESCRIPTION
**Pull request by owent**
Fixes #2676 
## Changes

+ Read default proto version from `third_party_release`
+ Format `cmake/opentelemetry-proto.cmake`

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: cmake/opentelemetry-proto.cmake [hunk 1]
```diff
@@ -43,7 +43,15 @@ else()
...
       STATUS "opentelemetry-proto dependency satisfied by: github download")
     if("${opentelemetry-proto}" STREQUAL "")
-      set(opentelemetry-proto "v1.0.0")
+      file(READ "${CMAKE_CURRENT_LIST_DIR}/../third_party_release"
+           OTELCPP_THIRD_PARTY_RELEASE_CONTENT)
+      if(OTELCPP_THIRD_PARTY_RELEASE_CONTENT MATCHES
+         "opentelemetry-proto=[ \\t]*([A-Za-z0-9_\\.\\-]+)")
+        set(opentelemetry-proto "${CMAKE_MATCH_1}")
+      else()
+        set(opentelemetry-proto "v1.1.0")
```
Comment (marcalff) [inline cmake/opentelemetry-proto.cmake:52]:
```suggestion
        set(opentelemetry-proto "v1.3.1")
```

**Review by marcalff — APPROVED** [review]
Thanks for the fix.

See suggested change.

Not sure why opentelemetry-proto.cmake was not formatted in my previous change, as CI was ok on format.

**Review by marcalff — CHANGES_REQUESTED** [review]
Also, please update third_party_release to use opentelemetry-proto 1.3.1

(Edit: this is something missing from PR #2669)

File: cmake/opentelemetry-proto.cmake (hunk 1 above)
Comment (owent) [reply cmake/opentelemetry-proto.cmake]: Done

**Comment by owent** [comment]
> Thanks for the fix.
> 
> See suggested change.
> 
> Not sure why opentelemetry-proto.cmake was not formatted in my previous change, as CI was ok on format.

Maybe we should use `cmake-format -i $($FIND -name 'CMakeLists.txt' -print -o -name '*.cmake' -print -o -name '*.cmake.in' -print)` in `tools/format.sh`. (Add `-o` between rules.)

**Review by marcalff — APPROVED** [review]
LGTM, thanks for the fix.

**Pull request merged**